### DISCUSSION TITLE
Why don't we use the .jsx file name extension instead of .js when creating a new app with create-next-app?

### DISCUSSION THREAD

**Discussion by adriancuadrado** [vercel/next.js#63286] [Help]
### Summary

.js works for react components but, isn't .jsx just more appropriate?

### Additional information

I was about to create an issue for this but then found this one that was already created a while ago in January's 8th: https://github.com/vercel/next.js/issues/60391

@leerob added [this comment](https://github.com/vercel/next.js/pull/60392#issuecomment-1902483600) in the associated pull request saying "It's okay to keep this as .js 👍" and I'm confused because yeah, its ok to keep the .js extension because it works, but wouldn't it bevbetter to use .jsx instead?

**Comment by icyJoseph** [comment]
This is an age old discussion. 

Whether or not the file extension should convey more meaning. React has always been on the, it is just JavaScript ™️ side of things. 

At the end of the day it is perhaps a matter of preference, but given that a JSX element is almost just make belief at the file text level, one could argue that JSX files are just JavaScript files. You still need to bring in a pragma to consume the JSX into a call to createElement and what not. 

I think, this take is very insightful too: 

- https://github.com/airbnb/javascript/pull/985#issuecomment-239145468

That being said, you might say, but what about TSX, and the reason for that one is that TypeScript type annotations do relay on `<>` and a parsing rule is necessary: https://www.typescriptlang.org/docs/handbook/jsx.html#the-as-operator
